agent/src/com/jonnyzzz/teamcity/virtual/run/docker/DockerVM.java

```diff
@@ -147,7 +163,7 @@ public BuildProcess startImpl() throws RunBuildException {
                 "--name=" + name,
                 "-v",
                 checkoutDir.getPath() + ":" + mountPoint + ":" + ctx.getDockerMountMode(),
-                "--workdir=" + mountPoint + "/" + RelativePaths.resolveRelativePath(checkoutDir, workDir),
+                "--workdir=" + mountPoint + ctx.getPathSeparatorInsideContainer() + RelativePaths.resolveRelativePath(checkoutDir, workDir),
```

Comment:
Let's check how it would work with Docker. If not we should warn we cannot use it in the build and and/or in the runner settings UI